tests/std/tests/VSO_0000000_regex_use/test.cpp

```diff
@@ -669,6 +669,43 @@ void test_gh_5160() {
     neg_regex.should_search_fail(L"xxxYxx\x2009xxxZxxx"); // U+2009 THIN SPACE
 }
 
+void test_gh_5192() {
+    // GH-5192: Correct characters not matched by special character dot
+    using namespace string_literals;
+    for (const syntax_option_type option : {
+             regex_constants::basic,
+             regex_constants::extended,
+             regex_constants::awk,
+             regex_constants::grep,
+             regex_constants::egrep,
+         }) {
+        const test_regex regex(&g_regexTester, "^.*", option);

```

Comment:
The variable names `regex` and `wregex` below are shadowing the Standard types. Shadowing is intensely confusing and interferes with searching, so we should always avoid it.

(I've stopped complaining when I see `expected`, `count`, etc. but this is a `<regex>` test so we need to avoid shadowing its contents.)